lld/MachO/Driver.cpp

```diff
@@ -407,8 +407,28 @@ static InputFile *addFile(StringRef path, LoadType loadType,
   case file_magic::macho_dynamically_linked_shared_lib_stub:
   case file_magic::tapi_file:
     if (DylibFile *dylibFile =
-            loadDylib(mbref, nullptr, /*isBundleLoader=*/false, isExplicit))
+            loadDylib(mbref, nullptr, /*isBundleLoader=*/false, isExplicit)) {
+      if (isExplicit && !dylibFile->allowableClients.empty()) {
```

Comment:
A bit bikesheddy: I think this logic probably belongs in loadDylib(), or even the dylib constructor. That's where other similar changes are, right?

Reply:
I looked at that initially, but we would need to pass the client name to `loadDylib` or the constructor, and that seemed a bit much for me.

Would you rather I moved it there?

Reply:
The client name is in `config` which is accessible globally, no?

Reply:
Ah yes, good point. Let me move this.

Reply:
Done.